src/plugins/console/public/lib/mappings/mappings.ts

```diff
@@ -307,53 +358,52 @@ export function clearSubscriptions() {
   }
 }
 
-const retrieveMappings = async (http, settingsToRetrieve, dataSourceId) => {
-  const { body: mappings } = await retrieveSettings(
-    http,
-    'fields',
-    settingsToRetrieve,
-    dataSourceId
-  );
-  if (mappings) {
+// Type guard function
+function isHttpResponse<T>(object: any): object is HttpResponse<T> {
+  return object && typeof object === 'object' && 'body' in object;
+}
```

Comment:
This is type guard for functions that receive http responses, because without properly checking the typescript produces error.